### PR TITLE
Remove forbidden `@author` tag

### LINKED ISSUE #37245: [Issue] Remove forbidden `@author` tag
This issue is automatically created based on existing pull request: magento/magento2#36981: Remove forbidden `@author` tag

---------
### Description
According to https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html#documentation-space, the `@author` tag is not permitted in Magento. This pull request removes this tag from some modules. Given there are so many instances of this tag, I've opened a small pull request to get the process started. I expect that the linter will force me to fix several other coding standards violations on the way, so having a smaller pull request means that task is easier to manage. I plan to open more pull requests to tackle the other instances of this tag.

See also https://github.com/magento/magento-coding-standard/pull/382 and https://github.com/magento/magento-coding-standard/issues/167

See also https://github.com/magento/magento2/pull/36304, https://github.com/magento/magento2/pull/36976, https://github.com/magento/magento2/pull/36977, https://github.com/magento/magento2/pull/36978, https://github.com/magento/magento2/pull/36979, https://github.com/magento/magento2/pull/36980

### Manual testing scenarios
There are not code changes in this pull request. This pull request only removes forbidden comments.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by fredden**
### Description
According to https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html#documentation-space, the `@author` tag is not permitted in Magento. This pull request removes this tag from some modules. Given there are so many instances of this tag, I've opened a small pull request to get the process started. I expect that the linter will force me to fix several other coding standards violations on the way, so having a smaller pull request means that task is easier to manage. I plan to open more pull requests to tackle the other instances of this tag.

See also https://github.com/magento/magento-coding-standard/pull/382 and https://github.com/magento/magento-coding-standard/issues/167

See also https://github.com/magento/magento2/pull/36304, https://github.com/magento/magento2/pull/36976, https://github.com/magento/magento2/pull/36977, https://github.com/magento/magento2/pull/36978, https://github.com/magento/magento2/pull/36979, https://github.com/magento/magento2/pull/36980

### Manual testing scenarios
There are not code changes in this pull request. This pull request only removes forbidden comments.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37245: Remove forbidden `@author` tag

### CODE REVIEW

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run Integration Tests

File: app/code/Magento/User/Model/ResourceModel/User/Locked/Collection.php [hunk 1]
```diff
@@ -22,5 +20,7 @@ protected function _initSelect()
     {
         parent::_initSelect();
         $this->addFieldToFilter('lock_expires', ['notnull' => true]);
+
+        return $this;
```
Comment (leonhelmus) [inline app/code/Magento/User/Model/ResourceModel/User/Locked/Collection.php:24]: Are you sure about this change? Im not sure if the docblock was still current?

File: app/code/Magento/Weee/Block/Item/Price/Renderer.php [hunk 2]
```diff
@@ -6,7 +6,7 @@
 namespace Magento\Weee\Block\Item\Price;
 
 use Magento\Framework\Pricing\PriceCurrencyInterface;
-use Magento\Sales\Model\Order\CreditMemo\Item as CreditMemoItem;
+use Magento\Sales\Model\Order\Creditmemo\Item as CreditMemoItem;
```
Comment (leonhelmus) [inline app/code/Magento/Weee/Block/Item/Price/Renderer.php:9]: Nice change! Wasn't this like a BIG issue?

**Review by leonhelmus — APPROVED** [review]

**Comment by leonhelmus** [comment]
@magento create issue

**Comment by ihor-sviziev** [comment]
Adding the same priority as https://github.com/magento/magento2/pull/36976#event-8742638190

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**

-  Verified @author tag got removed from `Magento_url-user-variable-wee`module.

**Before:** :heavy_multiplication_x: 

Screenshot 2023-03-28 at 11 23 31 AM

Screenshot 2023-03-28 at 11 23 51 AM
Screenshot 2023-03-28 at 11 24 26 AM


**After:** :heavy_check_mark:  


Screenshot 2023-03-28 at 11 27 08 AM
Screenshot 2023-03-28 at 11 27 15 AM


No addiitional manual test cases is required for the regression as this PR related to the removal of the @author tag from the `Magento_url-user-variable-wee` module.

Multiple builds are failing hence, moving this PR to **Extended Testing**

File: app/code/Magento/Weee/Block/Item/Price/Renderer.php (hunk 2 above)
Comment (fredden) [reply app/code/Magento/Weee/Block/Item/Price/Renderer.php:9]: This showed up in the PHP Mess Detector check, which is run as part of the automated Static Tests check on GitHub. I don't know what this impacts, but there were no unit tests failures due to this bug/change.

File: app/code/Magento/User/Model/ResourceModel/User/Locked/Collection.php (hunk 1 above)
Comment (fredden) [reply app/code/Magento/User/Model/ResourceModel/User/Locked/Collection.php:24]: Other similar methods return self. Eg, https://github.com/magento/magento2/blob/a047c7639c63bf2b858e0bbd9689eb993a48a5a2/app/code/Magento/Customer/Model/ResourceModel/Grid/Collection.php#L73-L79
https://github.com/magento/magento2/blob/a047c7639c63bf2b858e0bbd9689eb993a48a5a2/app/code/Magento/Review/Model/ResourceModel/Review/Product/Collection.php#L167-L172

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run Functional Tests B2B, Functional Tests CE ,Functional Tests EE

**Comment by engcom-Delta** [comment]
Functional Tests B2B  failure is different on last two run
run1
Screenshot 2023-03-30 at 9 42 15 AM
run2
Screenshot 2023-03-30 at 9 43 20 AM
Hence, Moving this PR to merge in Progress.

**Pull request merged**